client/src/main/java/io/pravega/client/stream/TransactionalEventStreamWriter.java

```diff
@@ -26,17 +26,19 @@
      * Start a new transaction on this stream. This allows events written to the transaction be written an committed atomically.
      * Note that transactions can only be open for {@link EventWriterConfig#transactionTimeoutTime}.
      * 
+     * @throws TxnFailedException This exception could be thrown on beginTxn
      * @return A transaction through which multiple events can be written atomically.
      */
-    Transaction<Type> beginTxn();
+    Transaction<Type> beginTxn() throws TxnFailedException;
```

Comment:
How can a transaction that does not exist can fail?
Maybe it is fine to have this error here but it deserves a little explanation in javadocs

Reply:
Yes Enrico